lib/Sema/TypeCheckDeclObjC.cpp

```diff
@@ -3614,6 +3812,18 @@ class ObjCImplementationChecker {
     }
   }
 
+  static SmallString<64> makeObjCAttrInsertionText(Identifier categoryName) {
```

Comment:
Nitpick: is there a way we could share the two copies of `makeObjCAttrInsertionText`?

Reply:
Yeah, I was being a little lazy here. Let me see if I can clean this up.

Reply:
Thank you!

Reply:
Took some doing, but I've generalized `fixDeclarationObjCName()` so I can apply it to extensions instead of writing a one-off helper for objcImpl to use.